### PR TITLE
Rewrites IBM tutorials to match QSCOUT tutorial template

### PR DESCRIPTION
**Pull request by dowusu-antwi**
See `qss` and `css` notebooks in `docs/source/optimizations/ibm`, e.g., `ibm_compile_css.ipynb`:
image

### CODE REVIEW

File: docs/source/optimizations/ibm/ibmq_compile_css.ipynb [hunk 1]
```diff
@@ -22,21 +22,27 @@
    ]
```
Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibmq_compile_css.ipynb:22]: <p>Typo: should be passed as an argument to css.Service()</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibmq_compile_css.ipynb:22]: <p>This should be the second circuit: </p><pre class="ql-syntax" spellcheck="false">Line #1.    # Visualize the pulse sequence for the second compiled circuit
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

File: docs/source/optimizations/ibm/ibmq_compile_qss.ipynb [hunk 2]
```diff
@@ -21,9 +21,37 @@
     "</a>"
```
Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: <p>Maybe worth reducing the output here as well to match the css notebook version:</p><pre class="ql-syntax" spellcheck="false">%pip install -q qiskit-superstaq[examples]
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: <p>Should we just use qss.SuperstaQProvider() to stay close to the css.Service() in the css notebook?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Compile with ibmq compile
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: <p>Following comment from the css notebook:</p><pre class="ql-syntax" spellcheck="false">Line #1.    # Visualize the pulse sequence for the second compiled circuit
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/SupertechLabs/client-superstaq/pull/539/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

**Comment by bharat-thotakura** [comment]
I think the notebooks are pretty much in a complete state after some typo fixes (and I think the addendum was a helpful touch!)

File: docs/source/optimizations/ibm/ibmq_compile_css.ipynb (hunk 1 above)
Comment (dowusu-antwi) [reply docs/source/optimizations/ibm/ibmq_compile_css.ipynb:22]: Fixed.

Comment (dowusu-antwi) [reply docs/source/optimizations/ibm/ibmq_compile_css.ipynb:22]: Fixed.

File: docs/source/optimizations/ibm/ibmq_compile_qss.ipynb (hunk 2 above)
Comment (dowusu-antwi) [reply docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: Done.

Comment (dowusu-antwi) [reply docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: Good catch -- done.

Comment (dowusu-antwi) [reply docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: Thanks, fixed.

Comment (dowusu-antwi) [reply docs/source/optimizations/ibm/ibmq_compile_qss.ipynb]: Fixed.

**Comment by bharat-thotakura** [comment]
LGTM after [3d6b4d2](https://github.com/SupertechLabs/client-superstaq/pull/539/commits/3d6b4d2c7b886ab505b1b3904124780dc81eb201). Will leave final approval to @knsmith @stephanielee9 in case they have further comments.

**Comment by stephanielee9** [comment]
Overall, looks good! A couple things I think we need to update:
* It would be nice if the `ibm.rst` file had a brief description - doesn't need to be very lengthy (could be a couple sentences or a small paragraph) or have references but Bharath put together a nice one for the [QSCOUT tutorial](https://superstaq.readthedocs.io/en/latest/optimizations/qscout/qscout.html)
* The Colab and Binder linked images don't work in the build - [this](https://github.com/SupertechLabs/client-superstaq/issues/525) should work: `[alt text](image_url)](link_url)`
* Nit: "Let us start by creating an example qiskit circuit" <-- that sentence in both notebooks. Should be Cirq in the `css` notebook and maybe capitalize or put in in-line code for the `qss` notebook.

**Comment by dowusu-antwi** [comment]
> It would be nice if the `ibm.rst` file had a brief description - doesn't need to be very lengthy (could be a couple sentences or a small paragraph) or have references but Bharath put together a nice one for the [QSCOUT tutorial](https://superstaq.readthedocs.io/en/latest/optimizations/qscout/qscout.html)

Done.

> The Colab and Binder linked images don't work in the build - [this](https://github.com/SupertechLabs/client-superstaq/issues/525) should work: `[alt text](image_url)](link_url)`

Fixed.

> Nit: "Let us start by creating an example qiskit circuit" <-- that sentence in both notebooks. Should be Cirq in the `css` notebook and maybe capitalize or put in in-line code for the `qss` notebook.

Fixed (capitalized).

**Review by stephanielee9 — APPROVED** [review]
lgtm!!

File: docs/source/optimizations/ibm/ibm.rst [hunk 3]
```diff
@@ -1,8 +1,19 @@
 SuperstaQ for IBM
```
Comment (bharat-thotakura) [inline docs/source/optimizations/ibm/ibm.rst:1]: nit: 
```suggestion
Superstaq for IBM
```

**Review by bharat-thotakura — APPROVED** [review]

**Pull request merged**